src/sentry/tasks/post_process.py

```diff
@@ -692,8 +733,7 @@ def process_event(data: dict, group_id: Optional[int]) -> Event:
 
     # Re-bind node data to avoid renormalization. We only want to
     # renormalize when loading old data from the database.
-    event.data = EventDict(event.data, skip_renormalization=True)
-
+    setattr(event, "data", EventDict(event.data, skip_renormalization=True))
```

Comment:
I think the Field set type can be expanded to allow this assignment

Reply:
could you elaborate what you mean by that? i

Reply:
the field is probably:

```
data: models.Fields[dict[str, Any], dict[str, Any]]
```

I forget if it's the first or second generic parameter but one of those is the "set" type -- which could be expanded to a `Mapping` of sorts which allows `EventDict`

Reply:
ahh, the setter already takes a `Mapping`:
```py
@data.setter
    def data(self, value: Mapping[str, Any]) -> None:
```

Reply:
maybe try this patch?
```diff
diff --git a/src/sentry/models/event.py b/src/sentry/models/event.py
index 01fbc4d9a7..494cb6308f 100644
--- a/src/sentry/models/event.py
+++ b/src/sentry/models/event.py
@@ -1,10 +1,11 @@
+from typing import Any
 from sentry_relay.processing import StoreNormalizer
 
 from sentry.db.models import NodeData
 from sentry.utils.canonical import CanonicalKeyDict
 
 
-class EventDict(CanonicalKeyDict):
+class EventDict(CanonicalKeyDict[str, Any]):
     """
     Creating an instance of this dictionary will send the event through basic
     (Rust-based) type/schema validation called "re-normalization".
 ```

what's the error it complains about with the normal assignment?

Reply:
`Incompatible types in assignment (expression has type "EventDict", variable has type "NodeData")  [assignment]`

Reply:
error: `Incompatible types in assignment (expression has type "EventDict", variable has type "NodeData")  [assignment]`
relevant line in `post_process.py`: `event.data = EventDict(event.data, skip_renormalization=True)`

Reply:
i think it's related to this https://github.com/python/mypy/issues/3004

Reply:
ah... in that case let's `# type: ignore[assignment]  # python/mypy#3004` instead of tricking the checker

Reply:
thanks! i'll add that in and merge once ci finishes up 😄🚀